python/tests/core/test_preprocessing.py

```diff
@@ -682,6 +686,27 @@ def test_apply_list(
             assert X.shape == Y.shape
 
 
+def test_homogeneous_column() -> None:
+    d = {"col1": [1, 2, 3, 4], "col2": [5.0, 6.0, 7.0, 8.0], "col3": ["a", "b", "c", "d"]}
+    df = pd.DataFrame(data=d)
+    types = {"col1": (int, ColumnProperties.homogeneous), "col2": (float, ColumnProperties.default)}
```

Comment:
TODO: follow up with a test that has something like:
* "col2": float
* "col2: (float)